storage/innobase/row/row0merge.cc

```diff
@@ -1166,18 +1159,31 @@ dberr_t row_merge_buf_write(const row_merge_buf_t *buf,
 
 		if (blob_file) {
 			ut_ad(buf->index->is_primary());
-			err = row_merge_buf_blob(
-				entry, n_fields, &blob_heap, blob_file);
+			if (!blob_heap) {
+				blob_heap = mem_heap_create(100);
+			}
```

Comment:
How was the initial size 100 bytes determined? If it is too small, a subsequent `mem_heap_alloc()` might just waste that block and allocate a new one. If it is too large, the end of the block would go wasted.